File: 10. Prefix Sums/238. Product of Array Except Self/Solution.cpp
#include <vector>

class Solution {
public:
    std::vector<int> productExceptSelf(std::vector<int>& nums) {
        std::vector<int> res(nums.size(), 1);

        for (int i = 1; i < nums.size(); ++i)
        {
            res[i] = res[i - 1] * nums[i - 1];
        }

        int postfixProduct = 1;
        
        for (int j = nums.size() - 1; j > -1; --j)
        {
            res[j] *= postfixProduct;
            postfixProduct *= nums[j];
        }

        return res;
    }
};

File: 10. Prefix Sums/238. Product of Array Except Self/Solution.py
from typing import List

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [1] * n

        for i in range(1, n):
            res[i] = res[i - 1] * nums[i - 1]
        
        postfix_product = 1

        for j in range(n - 1, -1, -1):
            res[j] *= postfix_product
            postfix_product *= nums[j]
        
        return res